Userland/Libraries/LibGfx/ImageFormats/PNGWriter.cpp

```diff
@@ -281,4 +322,26 @@ ErrorOr<ByteBuffer> PNGWriter::encode(Gfx::Bitmap const& bitmap, Options options
     return ByteBuffer::copy(writer.m_data);
 }
 
+ErrorOr<ByteBuffer> PNGWriter::encodeAnimated(AK::Vector<Gfx::Bitmap const&> bitmap, u16 delay_num = 1, u16 delay_den = 30, u8 dispose_op = 0, u8 blend_op = 0, Options options)
+{
+    PNGWriter writer;
+    TRY(writer.add_png_header());
+    TRY(writer.add_IHDR_chunk(bitmap[0].width(), bitmap[0].height(), 8, PNG::ColorType::TruecolorWithAlpha, 0, 0, 0));
+    if (options.icc_data.has_value())
+        TRY(writer.add_iCCP_chunk(options.icc_data.value()));
+    TRY(writer.add_acTL_chunk(bitmap.size()));
+    // Static image is not part of animation
+    u32 sequence_number = 0;
+    TRY(writer.add_IDAT_chunk(bitmap[0], false, sequence_number));
+    for (auto& b : bitmap) {
+        dbgln("frame #{}", sequence_number);
+        TRY(writer.add_fcTL_chunk(sequence_number, b.width(), b.height(), delay_num, delay_den, dispose_op, blend_op));
+        sequence_number++;
+        TRY(writer.add_IDAT_chunk(b, true, sequence_number));
```

Comment:
This is writing `bitmap[0]` a second time. Is that intentional?

Reply:
APNG accepts two formats. In this, the thumbnail is separate from the first frame. In the other, it's the same as the first frame. I was originally going to allow the submission of a separate thumbnail image, but this probably has little purpose in this context. I might rework it a bit.